SimonAwiti/Store-manager-app-endpoints
# # 161273838 Endpoints with passing tests

#### What does this PR do?
* Contains all the sales records and products endpoints 
* Has passing tests integrated to Travis
## The following are API endpoints: 
* Get all available products
* Fetch a single product record
* Fetch all sale records 
* Create a product 
* Create a sale order
## Here is a list of the functioning endpoints

| EndPoint                                | Functionality                              |   
| :---                                         |     :---:                                         |  
| GET /products                       | Get all available products           |  
| GET /products/<productId> | Fetch a single product record    |  
| GET /sales                              | Fetch all sale records                 | 
| GET /sales/<saleId>              | Fetch a single sale record          | 
| POST /products                     | Create a product                        |  
| POST /sales                           | Create a sale order                     |
  
## Extra endpoints include 
* Deleting both the sales records and products
* Editing (PUT) both sales records and products

## Background context 
Published POSTMAN documentation
[Documentation](https://documenter.getpostman.com/view/5353857/RWgtTwtr#intro)
## Testing the endpoints
* Install python then using pip install.. install flask
* clone the repo
* Ensure that postman is installed
* From your terminal locate the repo and run: python run.py
* open postman and test the endpoints
* Use unit test to run the tests
## screenshots
![capture](https://user-images.githubusercontent.com/42590964/47090120-e6f3d900-d22a-11e8-9784-d1e18d0465b3.PNG)
![capture1](https://user-images.githubusercontent.com/42590964/47090135-ee1ae700-d22a-11e8-8e82-679fd7ceacd6.PNG)
![capture3](https://user-images.githubusercontent.com/42590964/47090145-f4a95e80-d22a-11e8-8dac-9ed9206981cd.PNG)
